tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,773 @@
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        'Sep',
+        'The current month is set correctly'
+    );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            viewMode="years"
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        '2020-2029',
+        'The year range is set correctly'
+    );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+});
+
+test( 'Navigating Forward by Month', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'September ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '.sl-calendar .next' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'October ' + currentYear,
+        'The next month is set correctly'
+    );
+});
+
+test( 'Navigating Backward by Month', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'September ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '.sl-calendar .prev' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The previous month is set correctly'
+    );
+});
+
+test( 'Navigating Forward by Year', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
```

Comment:
this.set( 'content', testContent ); is not needed for this test and has been removed